src/Common/Core/Impl/IO/IFileSystem.cs

```diff
@@ -28,5 +30,9 @@ public interface IFileSystem {
         void DeleteDirectory(string path, bool recursive);
         string[] GetFileSystemEntries(string path, string searchPattern, SearchOption options);
         void CreateDirectory(string path);
+
+        string CompressFile(string path);
+        string CompressDirectory(string path);
+        string CompressDirectory(string path, Matcher matcher, Action<string> callback);
```

Comment:
>callback [](start = 78, length = 8)

Since this is effectively a progress callback now, consider using IProgress<T> here for the sake of clarity. At call site, Progress<T> provides an easy way to wrap a lambda, so it should be almost as concise.